tests/expecter.go

```diff
@@ -0,0 +1,226 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+
+	expect "github.com/google/goexpect"
+	"google.golang.org/grpc/codes"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+)
+
+const (
+	PromptExpression = `(\$ |\# )`
+	CRLF             = "\r\n"
+)
+
+var (
+	shellSuccess = regexp.MustCompile(RetValue("0"))
+	shellFail    = regexp.MustCompile(RetValue("[1-9].*"))
+)
+
+// VmiConsoleExpectBatch runs the batch from `expected` connecting to the `vmi` console and
+// wait `timeout` for the batch to return.
+// NOTE: there is a safer version that validates sended commands `CheckForTextExpecter` refer to it about limitations.
+func VmiConsoleExpectBatch(vmi *v1.VirtualMachineInstance, expected []expect.Batcher, timeout time.Duration) error {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		return err
+	}
+
+	expecter, _, err := NewConsoleExpecter(virtClient, vmi, 30*time.Second)
+	if err != nil {
+		return err
+	}
+	defer expecter.Close()
+
+	resp, err := expecter.ExpectBatch(expected, timeout)
+	if err != nil {
+		log.DefaultLogger().Object(vmi).Infof("%v", resp)
+	}
+	return err
+}
+
+// CheckForTextExpecter runs the batch from `expected` connecting to a VMI's console and
+// wait `timeout` for each expected batch command to return, it also check that the sended commands arrives to console checking.
```

Comment:
Ohh missed this one as well :/ sorry.
"for the batch to return" instead of "for each expected batch command to return" like in the previous one.